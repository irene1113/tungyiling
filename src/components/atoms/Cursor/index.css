.cursor {
	position: fixed;
	height: 40px;
	width: 40px;
	border-radius: 50%;
	background-color: var(--primary);
	transform: translateX(-50%) translateY(-50%);
	transition: all 0.3s ease-out;
	pointer-events: none;
	z-index: 200;

	&.small {
		height: 15px;
		width: 15px;
		background-color: white;
	}
}
