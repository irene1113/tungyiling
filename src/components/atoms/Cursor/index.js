import React, { useEffect } from 'react';
import classnames from 'classnames';

import { useMousemove } from 'util/event';
import { useBoolean } from 'util/hook';

import { useMenu } from 'models/global';

import styles from './index.css';

const Cursor = ({ className }) => {
	const [{ open }] = useMenu();
	const mousePosition = useMousemove();

	const [isSmall, { setFalse, setTrue }] = useBoolean();

	useEffect(() => {
		if (open && mousePosition.x > document.documentElement.clientWidth * 0.9 && isSmall) {
			setFalse();
		} else if (open && mousePosition.x < document.documentElement.clientWidth * 0.9 && !isSmall) {
			setTrue();
		} else if (!open && isSmall) {
			setFalse();
		}
	}, [open, mousePosition]);

	return (
		<div
			className={classnames(styles.cursor, className, { [styles.small]: isSmall })}
			style={{ left: `${mousePosition.x}px`, top: `${mousePosition.y}px` }}
		/>
	);
};

export default Cursor;
