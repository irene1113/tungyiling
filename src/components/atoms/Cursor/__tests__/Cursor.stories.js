import React from 'react';
import { storiesOf } from '@storybook/react';

import Cursor from 'components/atoms/Cursor';

const stories = storiesOf('atoms/Cursor', module);

stories.add('__interactive', () => <Cursor />, {
	redux: {
		data: {
			global: {
				menu: {
					open: true,
				},
			},
		},
	},
});

stories.add('small', () => <Cursor />, {
	redux: {
		data: {
			global: {
				menu: {
					open: true,
				},
			},
		},
	},
});
