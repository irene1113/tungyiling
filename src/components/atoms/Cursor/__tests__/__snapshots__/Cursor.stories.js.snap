// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="cursor small"
    style="left: -100px; top: 100px;"
  />
</DocumentFragment>
`;

exports[`small 1`] = `
<DocumentFragment>
  <div
    class="cursor small"
    style="left: -100px; top: 100px;"
  />
</DocumentFragment>
`;
